bin/weewx/reportengine.py

```diff
@@ -25,55 +26,92 @@
 from weeutil.weeutil import to_bool
 import weewx.manager
 
+# spans of valid values for each CRON like field
+MINUTES = (0, 59)
+HOURS = (0, 23)
+DOM = (1, 31)
+MONTHS = (1, 12)
+DOW = (0, 6)
+# valid day names for DOW field
+DAY_NAMES = ('sun', 'mon', 'tue', 'wed', 'thu', 'fri', 'sat')
+# valid month names for month field
+MONTH_NAMES = ('jan', 'feb', 'mar', 'apr', 'may', 'jun',
+               'jul', 'aug', 'sep', 'oct', 'nov', 'dec')
+# map month names to month number
+MONTH_NAME_MAP = zip(('jan', 'feb', 'mar', 'apr',
+                      'may', 'jun', 'jul', 'aug',
+                      'sep', 'oct', 'nov', 'dec'), xrange(1, 13))
+# map day names to day number
+DAY_NAME_MAP = zip(('sun', 'mon', 'tue', 'wed',
+                    'thu', 'fri', 'sat'), xrange(7))
+# map CRON like nicknames to equivalent CRON like line
+NICKNAME_MAP = {
+    "@yearly": "0 0 1 1 *",
+    "@anually": "0 0 1 1 *",
+    "@monthly": "0 0 1 * *",
+    "@weekly": "0 0 * * 0",
+    "@daily": "0 0 * * *",
+    "@hourly": "0 * * * *"
+}
+# list of valid spans for CRON like fields
+SPANS = (MINUTES, HOURS, DOM, MONTHS, DOW)
+# list of valid names for CRON lik efields
+NAMES = ((), (), (), MONTH_NAMES, DAY_NAMES)
+# list of name maps for CRON like fields
+MAPS = ((), (), (), MONTH_NAME_MAP, DAY_NAME_MAP)
+
 # =============================================================================
 #                    Class StdReportEngine
 # =============================================================================
 
 class StdReportEngine(threading.Thread):
     """Reporting engine for weewx.
-    
+
     This engine runs zero or more reports. Each report uses a skin. A skin
     has its own configuration file specifying things such as which 'generators'
     should be run, which templates are to be used, what units are to be used,
-    etc.. 
+    etc..
     A 'generator' is a class inheriting from class ReportGenerator, that
-    produces the parts of the report, such as image plots, HTML files. 
-    
+    produces the parts of the report, such as image plots, HTML files.
+
     StdReportEngine inherits from threading.Thread, so it will be run in a
     separate thread.
-    
+
     See below for examples of generators.
     """
-    
-    def __init__(self, config_dict, stn_info, gen_ts=None, first_run=True):
-        """Initializer for the report engine. 
-        
+
+    def __init__(self, config_dict, stn_info, record=None, gen_ts=None, first_run=True):
```

Comment:
Would suggest adding the record parameter to the end, on the remote chance someone has called the initializer without using keywords